src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php

```diff
@@ -94,4 +95,22 @@ private function loadResources()
             unset($this->resourceFiles[$key]);
         }
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function warmUp($cacheDir)
+    {
+        if (null !== $this->options['cache_dir']) {
+            if (null !== $this->locale) {
+                $this->loadCatalogue($this->locale);
```

Comment:
The current locale can be not in the fallback locales (if the current is the default one). So I prefer to load the catalogue for the current locale and then for fallback locales.